src/routes/webhook/discord/interaction/confess.ts

```diff
@@ -205,12 +220,29 @@ export async function handleConfess(
     channelId: Snowflake,
     authorId: Snowflake,
     [option, ...options]: InteractionApplicationCommandChatInputOption[],
+    resolved: Resolved | null,
+    permissions: bigint,
 ) {
-    strictEqual(options.length, 0);
+    strict(options.length <= 1);
     strictEqual(option?.type, InteractionApplicationCommandChatInputOptionType.String);
     strictEqual(option.name, 'content');
+
+    let attachment = null;
+    const attachments = Object.values(resolved?.attachments ?? {});
+    // retrieve attachment if it exists, we don't actually do anything with the attachment option and assume the sole resolved.attachments entry is the attachment
+    if (options.length === 1) {
+        // check for permission to attach files
+        assert(hasAllPermissions(permissions, ATTACH_FILES));
```

Comment:
We should refrain from asserting here because the `register-slash-commands.js` script doesn't set the `default_member_permission` to be `ATTACH_FILES`—and we wouldn't want to anyway because this is an optional argument. Instead, we should check for the permission and then throw a message back to the user that they do not have the permission to send attachments to the channel.

Alternatively, we _could_ have a dedicated `/confess-with-attachment` command that _does_ have the `default_member_permission` set to `ATTACH_FILES` so that the `attachment` argument is now `required: true`. I personally like this solution because we punt the permission checks to Discord's UI.

Reply:
Hm, I personally lean more toward the former solution, where we move the check up to the interaction handler and check for an attachment _and then_ check for the permission.

From a UI perspective, I think creating a whole new command would be redundant and de-incentivize use compared to making `/confess` a one-stop shop for our users. Creating a whole new command also feels overly atomic in that sense.

Reply:
Okay, that's fair. I agree with that! 🚀